Libraries/dnYara.Interop/Interops/ObjRefHelper.cs

```diff
@@ -72,6 +48,28 @@ public static void ForEachStringInObjRef(IntPtr ref_obj, Action<string> action)
             }
         }
 
+        /// walks a variable-sized array of pointers of type T, marshalling and running a custom validation function on each iteration of the pointer
+        /// This is an abstraction around specialized loops like `ForEachYaraMetaInObjRef`
+        public static IEnumerable<T> EachStructOfTInObjRef<T>(IntPtr ref_obj, Func<T, bool> validityChecker) where T: struct {
```

Comment:
there are a few other places this 'unmarshal + validate' pattern could be used but I didn't want to go changing _everything_ for no reason.

Reply:
Perfect !